### ISSUE TITLE
Problems validating emails with dots (.)

### ISSUE THREAD

**Issue by DiegoEP** [ghiscoding/angular-validation#45]
When I try to put emails with dots, the validator passes when dots are in the beginning or in the end.

E.g.:
my@email.com.
.my@email.com.
....my@email.com....

**Comment by ghiscoding** [comment]
Thanks for catching this error, it seems to be so hard to find a proper regular expression that covers everything, no one as a perfect pattern for it. 

So anyway, here is a better solution that I have for now, it covers most of the RFC cases, except a string inside double quotes "some string". You can test the following in your console if you want.
```javascript
var pattern = /^[-\wа-яàáâãäåæçèéêëœìíïîðòóôõöøùúûñüýÿßÞďđ0-9#~!$%^&*_=+\/`\|}{\'?]+(\.[-\wа-яàáâãäåæçèéêëœìíïîðòóôõöøùúûñüýÿßÞďđ0-9#~!$%^&*_=+\/`\|}{\'?]+)*@([\wа-яàáâãäåæçèéêëœìíïîðòóôõöøùúûñüýÿßÞďđ0-9_][-\wа-яàáâãäåæçèéêëœìíïîðòóôõöøùúûñüýÿßÞďđ0-9_]*(\.[-\wа-яàáâãäåæçèéêëœìíïîðòóôõöøùúûñüýÿßÞďđ0-9_]+)*([\wа-яàáâãäåæçèéêëœìíïîðòóôõöøùúûñüýÿßÞďđ]+)|(\.[\wа-яàáâãäåæçèéêëœìíïîðòóôõöøùúûñüýÿßÞďđ]{2,6})|([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}))(:[0-9]{1,5})?$/i;
var test1 = pattern.test("my@email.com."); 			// false
var test2 = pattern.test(".my@email.com."); 		// false
var test3 = pattern.test("....my@email.com...."); 	// false
var test4 = pattern.test("g$g.com"); 				// false
var test5 = pattern.test("g@g,com"); 				// false
var test6 = pattern.test("some space@hotmail.com"); // false
var test7 = pattern.test("someone@hotmail.com"); 	// true
var test8 = pattern.test("nickname@domain"); 		// true
var test9 = pattern.test("кокер@спаниель.рф"); 	   // true (russian characters)
var test10 = pattern.test("#!$%&'*+-/=?^_`{}|~@example.org"); 			// true
var test11 = pattern.test("other.email-with-dash@some-company.com"); 	// true
var test12 = pattern.test("hola.àáâãäåæçèéêëœìíïîðòóôõöøùúûñüýÿ@español.com"); // true (latin characters)

// show results in console
console.log(test1); // false  my@email.com.
console.log(test2); // false  .my@email.com.
console.log(test3); // false  ....my@email.com....
console.log(test4); // false  g$g.com
console.log(test5); // false  g@g,com
console.log(test6); // false  some space@hotmail.com
console.log(test7); // true   someone@hotmail.com
console.log(test8); // true   nickname@domain
console.log(test9); // true   кокер@спаниель.рф
console.log(test10); // true  #!$%&'*+-/=?^_`{}|~@example.org
console.log(test11); // true  other.email-with-dash@some-company.com
console.log(test12); // true  hola.àáâãäåæçèéêëœìíïîðòóôõöøùúûñüýÿ@español.com

// this should be valid but I can't seem to find the best way to modify my regular expression to accept double quotes and spaces inside it as shown below
// this should be valid:   "very.unusual.@.unusual.com"@example.com
// but my regular expression does not permit it yet
```
So apart from the double quotes ("), I have most of the other email that should be valid following the [Wikipedia - email address](https://en.wikipedia.org/wiki/Email_address#Examples)

I am also in the process of changing some piece of my code on how I evaluate regex, so it will take me a few more days before I finish everything.

**Comment by ghiscoding** [comment]
Fixed in latest revision #[1.3.35](https://github.com/ghiscoding/angular-validation/releases/tag/v1.3.35).

Thanks for the feedback :)
